### PR TITLE
Implement UT for OSBuildEvnConfig controller

### PR DESCRIPTION
**Pull request by ygalblum**
Update
------
- Finalizer
- Certificate
- psql
- ConfigMap for composer configuration
- ConfigMap for proxy configuration
- Composer deployment
- Composer services

Cleanup
-------
- Finalizer
- Certificate secret

### CODE REVIEW

File: controllers/osbuildenvconfig_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,675 @@
...
+		composerCertificateSecret = corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: operatorNamespace,
+				Name:      composerCertificateName,
+			},
+		}
+	)
+
+	BeforeEach(func() {
+		os.Setenv("WORKING_NAMESPACE", operatorNamespace)
```
[the remark this answers is not in the document]
Comment (ygalblum) [reply controllers/osbuildenvconfig_controller_test.go:89]: Done

File: controllers/osbuildenvconfig_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,740 @@
...
+			Scheme:                     scheme,
+			OSBuildEnvConfigRepository: osBuildEnvConfigRepository,
+			CertificateRepository:      certificateRepository,
+			ConfigMapRepository:        configMapRepository,
+			DeploymentRepository:       deploymentRepository,
+			ServiceRepository:          serviceRepository,
+			SecretRepository:           secretRepository,
+		}
+
+		requestContext = context.TODO()
```
Comment (bardielle) [inline controllers/osbuildenvconfig_controller_test.go:123]: you can use `request.Context()`

Comment (ygalblum) [reply controllers/osbuildenvconfig_controller_test.go:123]: Are you sure? I think you're confusing with another type of `request`

**Review by jakub-dzon — APPROVED** [review]
Please rebase.

File: controllers/osbuildenvconfig_controller_test.go (hunk 2 above)
Comment (bardielle) [reply controllers/osbuildenvconfig_controller_test.go:123]: by a mistake I thought `request` is from type "http.Request" 
So you can't use it like that

**Review by bardielle — APPROVED** [review]

**Pull request merged**